File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
# 반복문을 -로 타면 - 값은 인덱싱이 뒤로 간다.

def solution(elements):
    length=len(elements)
    arr=elements*2
    result =[]
    for i in range(length):
        ele=elements[i]
        result.append(elements[i])
        
        for j in range(i+1,i+length):
            ele+=arr[j]
            result.append(ele)
    result=set(result)
    return len(result)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        
        int len = elements.length;
        ArrayList<Integer> numbers = new ArrayList<>();
        for(int i =1;i<=len;i++){ // 길이
            
            for(int j=0;j<len;j++){ // 시작점
                int ans = 0;
                if(j+i>=len){ // 원형을 넘어선다면, 
                    int plus = j+i-len; // 앞에서부터 더해야할 갯수
                    for(int k=0;k<plus;k++){
                        ans+=elements[k];
                    }
                    for(int jj=j;jj<len;jj++){
                        ans+=elements[jj];
                    }
                    numbers.add(ans);
                }else{
                    for(int k=j;k<i+j;k++){
                        ans+=elements[k];
                    }
                }
                numbers.add(ans);
            }
            
        }
        HashSet<Integer> answer = new HashSet<>();
        for(Integer ans : numbers){
            answer.add(ans);
        }
        return answer.size();
    }
}

// int answer = 0;
//         ArrayList<Integer> array = new ArrayList<>();
//         for(int ele : elements){
//             array.add(ele);
//         }
        
//         System.out.println(array);